plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -455,25 +405,15 @@ public void onOutputLine(String line) {
       } else if (line.contains("Error: A fatal exception has occurred. Program will exit")) { //$NON-NLS-1$
         // terminate the Python process
         stop(false);
-      } else if ((matcher = moduleStartedPattern.matcher(line)).matches()
-          || (matcher = moduleRunningPattern.matcher(line)).matches()) {
+      } else if (shouldAutoDetectPort && (matcher = moduleRunningPattern.matcher(line)).matches()) {
         String serviceId = matcher.group("service");
         String url = matcher.group("url");
         moduleToUrlMap.put(serviceId, url);
+
         String portString = matcher.group("port");
         int port = parseInt(portString, 0);
-        if (port > 0 && (serverPortCandidate == 0 || "default".equals(serviceId))) { // $NON-NLS-1$
-          serverPortCandidate = port;
-        }
-      } else if ((matcher = adminStartedPattern.matcher(line)).matches()
-          || (matcher = adminRunningPattern.matcher(line)).matches()) {
-        int port = parseInt(matcher.group("port"), 0);
-        if (port > 0 && adminPort <= 0) {
-          adminPort = port;
-        }
-        // Admin comes after other modules, so no more module URLs
```

Comment:
For reviewers: this isn't the case with Dev App Server 1. It prints out an admin URL for each module, so we shouldn't be using this to determine that no more URLs are coming. We should just ignore admin URL lines for Dev App Server 1. I've also reworked the "find-a-candidate-and-assign-it-later" approach and made it more straightforward.